deployment/src/main/java/io/quarkiverse/githubaction/deployment/GitHubActionProcessor.java

```diff
@@ -387,19 +383,20 @@ private static void generatePayloadTypeResolver(ClassOutput beanClassOutput,
         payloadTypeResolverClassCreator.addAnnotation(Singleton.class);
 
         Map<String, DotName> payloadTypeMapping = eventDefinitions.stream()
-                .collect(Collectors.toMap(ed -> ed.getEvent(), ed -> ed.getPayloadType(), (ed1, ed2) -> ed1));
+                .collect(Collectors.toMap(EventDefinition::getEvent, EventDefinition::getPayloadType, (ed1, ed2) -> ed1));
 
         MethodCreator getPayloadTypeMethodCreator = payloadTypeResolverClassCreator.getMethodCreator("getPayloadType",
                 Class.class, String.class);
 
         ResultHandle eventRh = getPayloadTypeMethodCreator.getMethodParam(0);
 
         for (Entry<String, DotName> payloadTypeMappingEntry : payloadTypeMapping.entrySet()) {
-            BytecodeCreator matches = getPayloadTypeMethodCreator.ifTrue(getPayloadTypeMethodCreator.invokeVirtualMethod(
+            try (BytecodeCreator matches = getPayloadTypeMethodCreator.ifTrue(getPayloadTypeMethodCreator.invokeVirtualMethod(
```

Comment:
Let's not go that way. Unfortunately the API of `BytecodeCreator` is not ideal, we discussed that recently with the team.

They will be all closed by the `ClassCreator` when the class will be generated so we don't need to close them here - it might even be harmful.

Reply:
OK, I was missing the context here - reverted.